src/raise.cr

```diff
@@ -175,16 +175,21 @@ fun __crystal_get_exception(unwind_ex : LibUnwind::Exception*) : UInt64
   unwind_ex.value.exception_object
 end
 
-def raise(ex : Exception) : NoReturn
-  ex.callstack = CallStack.new
+# Raises the *exception*.
+#
+# Note that exception callstack will only be set if it's not already set.
+# It's useful for reraising of exceptions in `rescue` blocks.
```

Comment:
Maybe the doc could be simpler? Maybe:

> Raises an `Exception`.
>
> This will set the exception's callstack if it hasn't been already. Re-raising a previously catched exception won't replace the callstack.